src/assets/js/protocol/protocol-details.js

```diff
@@ -0,0 +1,229 @@
+            var unique = selector.replace(/[^a-zA-Z]+/g, '');
+            details.id = 'expand-' + unique + '-'+ _count;
+            _count++;
+        }
+
+        // close by default
+        // TODO: add support for open attribute
+        details.setAttribute('aria-hidden', true);
+        details.classList.add('is-closed');
+
+        // Control
+        control.setAttribute('type', 'button');
+        // add aria-controls
+        control.setAttribute('aria-controls', details.id);
+        // add aria-expanded
+        control.setAttribute('aria-expanded', false);
+        // add listener
+        control.addEventListener('click', Details.handleControlActivation, false);
+        // copy the summary's contents into the control
+        summary.childNodes.forEach(function(child) {
+            control.append(child);
+        });
+        // append control element
+        summary.appendChild(control);
+        summary.classList.add('is-summary');
+    };
+
+    /**
+     * destroyItem
+     * @param {Object} el - Element the control was placed inside of
+     * - does not attempt to remove the details wrapper
+     */
+    Details.destroyItem = function(el) {
+        var summary = el;
+        var parent = summary.parentNode;
+        var details = summary.nextElementSibling;
+        var control = summary.querySelector('button');
+        parent.classList.remove('is-details');
+        details.removeAttribute('aria-hidden');
+        details.classList.remove('is-closed');
+        // move control's contents back to summary
+        control.childNodes.forEach(function(child) {
+            summary.append(child);
+        });
+        summary.removeChild(control);
+        summary.classList.remove('is-summary');
+    };
+
+    /**
+     * Init
+     * @param {Object} selector - CSS selector matching "summary" elements
+     */
+    Details.init = function(selector) {
+        var summaries = doc.querySelectorAll(selector);
+        // loop through controls on the page and init them one at a time
+        for (var i = 0; i < summaries.length; i++) {
+            Details.initItem(summaries[i], selector);
+        }
+    };
+
+    /**
+     * Destroy
+     * @param {Object} selector - CSS selector matching "summary" elements
+     */
+    Details.destroy = function(selector) {
+        var summaries = doc.querySelectorAll(selector);
+        // loop through controls on the page and destroy them one at a time
+        for (var i = 0; i < summaries.length; i++) {
+            Details.destroyItem(summaries[i], selector);
+        }
+    };
+
+    // init as details/summary polyfill
+    if(!Details.isSupported) {
+        // not supported, add support
+        Details.init('summary');
+        // TODO are some aria things necissary with a passing feature test? tabindex? role=button?
+    }
+
+    // init generic class indicating headings should be made into expand/collapse componenet
+    Details.init('.mzp-c-details > h2');
```

Comment:
It would be good to write some JS tests for this component? This might mean separating out the calls to `init` in a separate file.